crates/precompiles/src/tip_fee_manager/mod.rs

```diff
@@ -393,11 +407,45 @@ impl TipFeeManager {
                 .ok_or(TempoPrecompileError::under_overflow())?,
         )?;
 
-        // If this is the first fee for the validator, record it in validators with fees
-        if collected_fees.is_zero() {
-            self.validator_in_fees_array.at(validator).write(true)?;
-            self.validators_with_fees.push(validator)?;
+        // Pre Allegro Moderato, add fees to be processed end of block
+        if !self.storage.spec().is_allegro_moderato() {
+            // If this is the first fee for the validator, record it in validators with fees
+            if collected_fees.is_zero() {
+                self.validator_in_fees_array.at(validator).write(true)?;
```

Comment:
are we setting `validator_in_fees_array` post AllegroModerato? the guard in `set_validator_token` still relies on it, a validator might accumulate fees in token x, then change to token y, then distribute fees in token y?